nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/storage/ITFetchAzureDataLakeStorage.java

```diff
@@ -16,43 +16,329 @@
+        String inputFlowFileContent = "InputFlowFileContent";
 
+        // WHEN
+        // THEN
+        testFailedFetch(fileSystem, directory, filename, inputFlowFileContent, inputFlowFileContent, 400);
     }
 
-    @Ignore
     @Test
-    public void testFetchFile() throws Exception {
+    public void testFetchNonExistentDirectory() {
+        // GIVEN
+        String directory = "TestDirectory";
+        String filename = "testFile.txt";
+        String inputFlowFileContent = "InputFlowFileContent";
+
+        // WHEN
+        // THEN
+        testFailedFetch(this.fileSystemName, directory, filename, inputFlowFileContent, inputFlowFileContent, 404);
+    }
+
+    @Test
+    public void testFetchNonExistentFile() {
+        // GIVEN
+        String directory = "TestDirectory";
+        String filename = "testFile.txt";
+        String inputFlowFileContent = "InputFlowFileContent";
+
+        fileSystemClient.createDirectory(directory);
+
+        // WHEN
+        // THEN
+        testFailedFetch(this.fileSystemName, directory, filename, inputFlowFileContent, inputFlowFileContent, 404);
+    }
+
+    @Ignore("Takes some time, only recommended for manual testing.")
+    @Test
+    public void testFetchLargeFile() {
+        // GIVEN
+        String directory = "TestDirectory";
+        String filename = "testFile.txt";
+        byte[] fileContentBytes = new byte[100_000_000];
+        String fileContent = new String(fileContentBytes);
+        String inputFlowFileContent = "InputFlowFileContent";
+
+        createDirectoryAndUploadFile(directory, filename, fileContent);
+
+        // WHEN
+        // THEN
+        testSuccessfulFetch(this.fileSystemName, directory, filename, inputFlowFileContent, fileContent);
+    }
+
+    @Test
+    public void testFetchInvalidDirectoryName() {
+        // GIVEN
+        String directory = "TestDirectory";
+        String invalidDirectoryName = "Test/\\Directory";
+        String filename = "testFile.txt";
+        String fileContent = "AzureFileContent";
+        String inputFlowFileContent = "InputFlowFileContent";
+
+        createDirectoryAndUploadFile(directory, filename, fileContent);
+
+        // WHEN
+        // THEN
+        testFailedFetch(this.fileSystemName, invalidDirectoryName, filename, inputFlowFileContent, inputFlowFileContent, 404);
+    }
+
+    @Test
+    public void testFetchInvalidFilename() {
+        // GIVEN
+        String directory = "TestDirectory";
+        String filename = "testFile.txt";
+        String invalidFilename = "test/\\File.txt";
+        String fileContent = "AzureFileContent";
+        String inputFlowFileContent = "InputFlowFileContent";
+
+        createDirectoryAndUploadFile(directory, filename, fileContent);
+
+        // WHEN
+        // THEN
+        testFailedFetch(this.fileSystemName, directory, invalidFilename, inputFlowFileContent, inputFlowFileContent, 404);
+    }
+
+    @Test
+    public void testFetchUsingExpressionLanguage() {
```

Comment:
It should also be tested when the expressions evaluated to empty string (eg. due to a missing attribute / variable).
It is interesting for the filesystem and the filename (the directory can be an empty string).
In these cases the processor should fail and it should not call the Azure service.